server/server.go

```diff
@@ -238,6 +238,8 @@ func (s *server) handle(client net.Conn) {
 	go func() {
 		var buf [1]byte
 		if _, err := io.ReadFull(client, buf[:]); err != nil {
+			close(sig)
+			client.Close()
```

Comment:
Nice catch. We should always close `sig`. How about changing the go function to always close sig via defer? In that case the second go-routine will do the client.Close always.

e.g.
```
go func() {
  defer close(sig)
  
  var buf [1]byte
  // keep rest as is
}()
```

Actually we might consider to remove the `client.Close` at the end, as the second go-routine in `handle` waits for a) the server shutting down or b) the handling go-routine having finished. If `sig` is closed properly we will always close the `client`.